nixos/modules/services/wayland/cage.nix

```diff
@@ -0,0 +1,82 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cage;
+in {
+  options.services.cage.enable = mkEnableOption ''
+    Enable the Cage Kiosk service.
+  '';
+
+  options.services.cage.user = mkOption {
+    type = types.str;
+    default = "demo";
+    description = ''
+      User to log-in as.
+    '';
+  };
+
+  options.services.cage.program = mkOption {
+    type = types.path;
+    default = "${pkgs.xterm}/bin/xterm";
+    description = ''
+      Program to run in cage.
+    '';
+  };
+
+  config = mkIf cfg.enable {
+
+    # The service is partially based off of the one provided in the
+    # cage wiki at
+    # https://github.com/Hjdskes/cage/wiki/Starting-Cage-on-boot-with-systemd.
+    systemd.services."cage@" = {
+      enable = true;
+      after = [ "systemd-user-sessions.service" "plymouth-quit.service" "dbus.socket" "systemd-logind.service" "getty@%i.service" "plymouth-deactivate.service" ];
+      before = [ "graphical.target" ];
+      wants = [ "dbus.socket" "systemd-logind.service" "plymouth-deactivate.service" ];
+      wantedBy = [ "graphical.target" ];
+      conflicts = [ "getty@%i.service" "plymouth-quit.service" "plymouth-quit-wait.service" ];
+
+      restartIfChanged = false;
+      serviceConfig = {
+        ExecStart = "${pkgs.cage}/bin/cage -d -- ${cfg.program}";
+        User = cfg.user;
+
+        # ConditionPathExists = "/dev/tty0";
+        IgnoreSIGPIPE = "no";
+
+        # Log this user with utmp, letting it show up with commands 'w' and
+        # 'who'. This is needed since we replace (a)getty.
+        UtmpIdentifier = "%I";
+        UtmpMode = "user";
+        # A virtual terminal is needed.
+        TTYPath = "/dev/%I";
+        TTYReset = "yes";
+        TTYVHangup = "yes";
+        TTYVTDisallocate = "yes";
+        # Fail to start if not controlling the virtual terminal.
+        StandardInput = "tty-fail";
+        StandardOutput = "syslog";
+        StandardError = "syslog";
+        # Set up a full (custom) user session for the user, required by Cage.
+        PAMName = "cage";
+      };
+    };
+
+    security.pam.services.cage.text = ''
+      auth    required pam_unix.so nullok
+      account required pam_unix.so
+      session required pam_unix.so
+      session required ${pkgs.systemd}/lib/security/pam_systemd.so
+    '';
+
+    systemd.targets.graphical.wants = [ "cage@tty1.service" ];
```

Comment:
On real hardware, I somehow had both `cage@graphical` and `cage@tty1` around. Did that happen for you as well?

Reply:
I didn't need cage@graphical. I would think that tty1 would be all that is needed. That said, I've tested it on virtualbox, so perhaps not the best example.

Reply:
I also didn't manually add it either - I thought something got confused by the template unit, but if you didn't see it, we can ignore it for now, and I'll do some more debugging later.

Reply:
I was able to reproduce this with your code as well - it seems systemd tries to start both `cage@tty1.service` and `cage@graphical.service`:

```
Feb 26 21:27:20 nixos systemd[1]: Started Login Service.
Feb 26 21:27:20 nixos systemd[1]: Reached target Multi-User System.
Feb 26 21:27:20 nixos systemd[1]: Started cage@graphical.service.
Feb 26 21:27:20 nixos systemd[1]: Started cage@tty1.service.
Feb 26 21:27:20 nixos systemd[1]: Reached target Graphical Interface.
Feb 26 21:27:20 nixos systemd[1]: Startup finished in 3.598s (kernel) + 4.674s (userspace) = 8.273s.
Feb 26 21:27:20 nixos systemd[730]: cage@graphical.service: Failed to set up standard input: No such file or directory
Feb 26 21:27:20 nixos systemd[730]: cage@graphical.service: Failed at step STDIN spawning /nix/store/9hf8zn44ji5xilkmdl7xr3s87ma7vlr3-cage-unstable-2020-01-18/bin/cage: No such file or directory
Feb 26 21:27:20 nixos systemd[1]: cage@graphical.service: Main process exited, code=exited, status=208/STDIN
Feb 26 21:27:20 nixos systemd[1]: cage@graphical.service: Failed with result 'exit-code'.
```

Reply:
In  #81327, I simply called the unit file `cage-tty1` and removed all the templating foo.